pkg/channel/event_receiver.go

```diff
@@ -251,19 +254,22 @@ func (r *EventReceiver) ServeHTTP(response nethttp.ResponseWriter, request *neth
 		response.WriteHeader(nethttp.StatusBadRequest)
 		return
 	}
-
+	polices, err := r.getPoliciesFor(channel.Namespace, channel.Name)
```

Comment:
we need to check, that `getPoliciesFor` is not nil (as this is set by the outside). As said in https://github.com/knative/eventing/pull/8123#discussion_r1709881861, we the could fail fast as the callback is not defined (log a error message and return with a 500 http code)